WordPress/gutenberg
# Block Editor: LinkControl: Incorporate settings in edit state

Closes #20007

This pull request seeks to resolve the issue described in #20007 where additional steps for editing links in paragraphs were introduced when refactoring the link component to use `LinkControl`. With these changes, the "Open in New Tab" toggle is always shown when viewing or editing a link. This allows for the behavior of <kbd>Cmd+K</kbd> to once again launch the user immediately into editing the URL of a link, where prior to these changes it required an additional step to switch from viewing to editing the link (clicking the "Edit" button).

&nbsp;|Previewing|Editing
---|---|---
Before|![Previewing before](https://user-images.githubusercontent.com/1779930/73999561-2029ee00-4933-11ea-81a7-0757f55f1b51.png)|![Editing before](https://user-images.githubusercontent.com/1779930/73999621-55364080-4933-11ea-8b17-a8813b843e93.png)
After|![Previewing after](https://user-images.githubusercontent.com/1779930/73973185-1fbc3380-48f0-11ea-84d9-a44b216fa396.png)|![Editing after](https://user-images.githubusercontent.com/1779930/73973261-3c586b80-48f0-11ea-9e89-9a4d93c676c9.png)

Also included:

- Replace "Reset" with "Submit"
- Variable-based styling in place of hard-coded
- Fixes selection placement when pressing <kbd>Escape</kbd>
- Prevents focus shifting back to paragraph after toggling "Open in New Tab"

**Testing Instructions:**

Confirm the following workflow goals for working with links in paragraphs:

1. Pressing <kbd>Cmd+K</kbd> will always allow you to immediately insert or edit the URL of a link
2. Pressing <kbd>Enter</kbd> after editing the URL will immediately restore your selection to the paragraph
3. You can toggle "Opens in New Tab" using only the keyboard
